### ISSUE TITLE
Question: What prevents podman to push layers to the registry during build?

### ISSUE THREAD

**Issue by StarpTech** [containers/podman#18642]
### Feature request description

Hi, building an image is a synchronous task. I'm curious if it would be possible to build and push layers to the registry simultaneously. If possible, I think this can save a lot of time because while layer two is built, layer 1 can be pushed to the registry.

```
podman build --layers --push --format v2s2 -t gcr.io/myrepo/myimage
```

**Comment by mtrmac** [comment]
This is interesting but also not something we are currently set up to do. c/image only provides an API to copy (“push”) completed images, not layers of unfinished images. And pushing a _complete_ image to a registry requires choosing a tag.

It would probably be possible to hack something together (choose a random long tag, hope that there is no collision; delete the tag afterwards), but that seems fairly messy. Pushing _just_ the layers as they are built would need to be a new API in c/image/copy, only for transports that opt in (like `docker://` could).

**Comment by StarpTech** [comment]
Naively, it can be implemented very similarly to how `--cache-from` and `--cache-to` is working with the important difference that pushing the intermediate layers isn't a blocking operation.

I experienced very slow builds when using podman distributed cache feature. Although layers are cached `Using cache ..` the layer was pushed again to the rInistry. At the end, it took more time or similiar time to build the image as without the cache.

**Comment by mtrmac** [comment]
Yeah, `--cache-to` is adding a tag to the repo, for each image pushed. So that would be user-visible.

**Comment by rhatdan** [comment]
@flouthoc could you help @StarpTech
